File: backend/src/main/java/com/spring/restaurantmanagementsystem/model/Store.java
package com.spring.restaurantmanagementsystem.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "stores")
@Getter @Setter @NoArgsConstructor
public class Store {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    @OneToMany(mappedBy = "store", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<User> users = new ArrayList<>();
}

File: frontend/src/types/store.ts
export interface Store {
  id: number;
  name: string;
  latitude: number;
  longitude: number;
}

export interface CreateStoreRequest {
  name: string;
  latitude: number;
  longitude: number;
}

export interface UpdateStoreRequest {
  name: string;
  latitude: number;
  longitude: number;
}
